pkg/api/validation/validation.go

```diff
@@ -1237,6 +1256,21 @@ func ValidatePersistentVolume(pv *api.PersistentVolume) field.ErrorList {
 			allErrs = append(allErrs, validateScaleIOVolumeSource(pv.Spec.ScaleIO, specPath.Child("scaleIO"))...)
 		}
 	}
+	if pv.Spec.LocalVolume != nil {
+		if numVolumes > 0 {
+			allErrs = append(allErrs, field.Forbidden(specPath.Child("localVolume"), "may not specify more than 1 volume type"))
+		} else {
+			numVolumes++
+			if !utilfeature.DefaultFeatureGate.Enabled(features.PersistentLocalVolumes) {
+				allErrs = append(allErrs, field.Forbidden(specPath.Child("localVolume"), "LocalVolumes are disabled by feature-gate"))
+			}
+			allErrs = append(allErrs, validateLocalVolumeSource(pv.Spec.LocalVolume, specPath.Child("localVolume"))...)
+			// StorageClass is required
+			if len(pv.Spec.StorageClassName) == 0 {
```

Comment:
@thockin yes, right now the idea is if the StorageClass has a topologyKey specified, then we will trigger this "delayed" binding behavior and also trigger the scheduler predicate to filter nodes based on topologyKey.

We could put the topologyKey in the PVC instead, but I prefer the idea of the StorageClass encapsulating the topology constraints so that the user does not need to be aware.

@ddysher we have no plans to change the unbinding flow.  It will operate the same way it does today.  PVC will be unbound when it is deleted.

Re-using the PVC could actually be a valid use case for data gravity.  You could have one pod that populates the PVC with data.  And another pod come along later to consume that PVC.